File: Tutorial/Examples/Basic_consumer/Consumer_Python/Consumer.py
import pika


def on_message_received(channe, method, properties, body):
    print(f'Received message: {body}')


# Set connection parameters. If connecting to a real server localhost should be replaced by the server´s address.
connection_parameters = pika.ConnectionParameters('localhost')

# Instantiate a connection unsing the connection_parameters previously defined
connection = pika.BlockingConnection(connection_parameters)

# Instantiate a channel
channel = connection.channel()

# Instantiate a queue
channel.queue_declare(queue='Transforms')

# Publish the message to the letterbox queue
channel.basic_consume(queue='Transforms', auto_ack=True,
                      on_message_callback=on_message_received)

print('Started consuming messages')

channel.start_consuming()

File: Tutorial/Examples/Basic_consumer/Consumer_C#/Consumer.cs
using System;
using System.Text;
using RabbitMQ.Client;
using RabbitMQ.Client.Events;

// Create connection.
var factory = new ConnectionFactory{ HostName = "localhost"};

// Instantiate the connection and one channel.
using var connection = factory.CreateConnection();
using var channel = connection.CreateModel();

// Declare a queue called Transforms.
channel.QueueDeclare(
    queue : "Transforms", 
    durable: false, 
    exclusive: false, 
    autoDelete: false);

// Declare consumer.
var consumer = new EventingBasicConsumer(channel);

// Subscribe to the Received event.
consumer.Received += (model, ea) =>{
    // Get the body of the message as byte array, decode to string and print it.
    var body = ea.Body.ToArray();
    var message = Encoding.UTF8.GetString(body);
    Console.WriteLine($"Message received: {message}");
};

// Start consuming messages from Transforms.
channel.BasicConsume(queue: "Transforms", autoAck: true, consumer: consumer);
Console.WriteLine("Started consuming");


// Wait for a keypress to exit.
Console.ReadKey();